lib/puppet/network/http_pool.rb

```diff
@@ -73,21 +74,25 @@ def self.http_ssl_instance(host, port, verifier = Puppet::SSL::Validator.default
   # @api public
   #
   def self.connection(host, port, use_ssl: true, ssl_context: nil)
-    if use_ssl
-      unless ssl_context
-        # TRANSLATORS 'ssl_context' is an argument and should not be translated
-        raise ArgumentError, _("An ssl_context is required when connecting to 'https://%{host}:%{port}'") % { host: host, port: port }
-      end
+    # When running in a JRuby master environment the Server manages our
+    # SSLContext for us.
+    if !Puppet::Util::Platform.jruby?
```

Comment:
The diff is a little confusing, but I think this change means the `connection` method is a noop on JRuby? If so, that will break puppetserver acting as an HTTPS client when submitting facts, etc to puppetdb. I'd expect the method to still call `http_client_class.new(host, port, ...)` where it calls the class that puppetserver sets.

Reply:
Yeah, that's ridiculous. I meant to condition the check/warning/error on JRuby's presence, not the whole creation of the client.